src/mantle/mantle_query_sync.c

```diff
@@ -75,3 +76,133 @@ GR_RESULT grWaitForFences(
         return GR_ERROR_OUT_OF_MEMORY;
     }
 }
+
+
+GR_RESULT grGetFenceStatus(
+    GR_FENCE fence)
+{
+    GrFence* grFence = (GrFence*)fence;
+    if (fence == NULL){
+	return GR_ERROR_INVALID_POINTER;
+    }
+
+    VkResult res = vki.vkGetFenceStatus(grFence->device, grFence->fence);
+    switch (res){
+    case VK_SUCCESS:
+	return GR_SUCCESS;
+    case VK_NOT_READY:
+	return GR_NOT_READY;
+    case VK_ERROR_DEVICE_LOST:
+	return GR_ERROR_DEVICE_LOST;
+    default:
+	break;
+    }
+    return GR_ERROR_UNKNOWN;
+}
+
+
+GR_RESULT grCreateEvent(
+    GR_DEVICE device,
+    const GR_EVENT_CREATE_INFO* pCreateInfo,
+    GR_EVENT* pEvent)
+{
+    GrDevice* grDevice = (GrDevice*)device;
+    if (grDevice == NULL || pCreateInfo == NULL || pEvent == NULL){
```

Comment:
should return `GR_ERROR_INVALID_HANDLE` when grDevice is null

Reply:
fixed